packages/bot-cleaning/src/MarketCleaner.ts

```diff
@@ -149,27 +152,37 @@ export class MarketCleaner {
         this.#whitelistedDustCleaningMaker &&
         this.#whitelistedDustCleaningMaker.has(offer.maker.toLowerCase())
       ) {
+        if (!offer.price) {
+          continue;
+        }
+
         const inboundTkn =
           ba == "bids" ? semibook.market.base : semibook.market.quote;
         const outboundTkn =
           ba == "bids" ? semibook.market.quote : semibook.market.base;
 
-        let takerWants = new Big(1).div(new Big(10).pow(outboundTkn.decimals));
-        let takerGives = inboundTkn.fromUnits(
-          takerWants.mul(offer.price!).toString()
-        );
-        if (outboundTkn.decimals > inboundTkn.decimals) {
-          takerGives = new Big(1).div(new Big(10).pow(inboundTkn.decimals));
-          takerWants = takerGives.mul(new Big(1).div(offer.price!).round());
+        let price = offer.price;
+        if (ba === "bids") {
+          price = new Big(1).div(offer.price);
+        }
+
+        /* this logic find the minimum takerGives */
+        const minPossibleWantsVolume = inboundTkn.fromUnits(1);
+        let minGivesVolume = outboundTkn.fromUnits("1");
+        let minWantsVolume = price.mul(minGivesVolume);
+
+        if (minWantsVolume.lt(minPossibleWantsVolume)) {
+          minWantsVolume = minPossibleWantsVolume;
+          minGivesVolume = new Big(1).div(price).mul(minWantsVolume);
         }
```

Comment:
This is a bit convoluted,  would be nice to have a comment on what this code is aiming to achieve and why

Reply:
I added a comment in strategies.ts and added 4 unit test :)